src/ra_wm_handler.h

```diff
@@ -0,0 +1,132 @@
+
+    _NORETURN void OnSetFocus(_In_ HWND hwnd, _In_ HWND hwndOldFocus) noexcept;
+    _NORETURN void OnKillFocus(_In_ HWND hwnd, _In_ HWND hwndNewFocus) noexcept;
+
+    _NORETURN void OnSetRedraw(_In_ HWND hwnd, _In_ BOOL fRedraw) noexcept;
+    _NORETURN void OnShowWindow(_In_ HWND hwnd, _In_ BOOL fShow, _In_ UINT status) noexcept;
+#pragma endregion
+
+    _NODISCARD UINT OnGetDlgCode(_In_ HWND hwnd, _In_ LPMSG lpmsg) noexcept;
+    _NODISCARD HWND OnNextDlgCtl(_In_ HWND hwnd, _In_ HWND hwndSetFocus, _In_ BOOL fNext) noexcept;
+    _NORETURN void OnEnterIdle(_In_ HWND hwnd, _In_ UINT source, _In_ HWND hwndSource) noexcept;
+
+    _NORETURN void OnNCPaint(_In_ HWND hwnd, _In_ HRGN hrgn) noexcept;
+    _NODISCARD BOOL OnEraseBkgnd(_In_ HWND hwnd, _In_ [[maybe_unused]] HDC hdc) noexcept;
+
+
+    // These normally should throw if the result is 0, but we aren't doing that right now
+#pragma region Common Button Handlers
+    // For non-modal dialogs and other window types, use OnClose
+    _NORETURN virtual void OnCloseModal(_In_ HWND hDlg) noexcept;
+    _NORETURN virtual void OnOK(_In_ HWND hDlg) noexcept;
+    _NORETURN virtual void OnCancel(_In_ HWND hDlg) noexcept;
+    _NORETURN virtual void OnAbort(_In_ HWND hDlg) noexcept;
+    _NORETURN virtual void OnRetry(_In_ HWND hDlg) noexcept;
+    _NORETURN virtual void ButtonCheck(_In_ HWND hwnd) noexcept;
+    _NORETURN virtual void ButtonUncheck(_In_ HWND hwnd) noexcept;
+
+    // This function will gray-out a check box or radio-button
+    _NORETURN virtual void DisableButtonCheck(_In_ HWND hwnd) noexcept;
+#pragma endregion
+
+
+};
+
+// Might be useful later, but don't understand it well enough yet -SBS
+#pragma region ClipboardHelper
+//struct ClipboardHelper : public WM_Handler
+//{
+//#pragma region Constructors
+//    ClipboardHelper() noexcept = default;
+//    virtual ~ClipboardHelper() noexcept = default;
+//    ClipboardHelper(const ClipboardHelper&) = delete;
+//    ClipboardHelper& operator=(const ClipboardHelper&) = delete;
+//    inline constexpr ClipboardHelper(ClipboardHelper&&) noexcept = default;
+//    inline constexpr ClipboardHelper& operator=(ClipboardHelper&&) noexcept = default;
+//#pragma endregion
+//    // https://docs.microsoft.com/en-us/windows/desktop/dataxchg/clipboard-overviews
+//#pragma region Methods
+//    _NORETURN virtual void OnCut(_In_ HWND hwnd) noexcept;
+//    _NORETURN virtual void OnCopy(_In_ HWND hwnd) noexcept;
+//    _NORETURN virtual void OnPaste(_In_ HWND hwnd) noexcept;
+//    _NORETURN virtual void OnClear(_In_ HWND hwnd) noexcept;
+//    _NORETURN virtual void OnDestroyClipboard(_In_ HWND hwnd) noexcept;
+//    _NORETURN virtual void OnDrawClipboard(_In_ HWND hwnd) noexcept;
+//#pragma endregion
+//
+//};  
+#pragma endregion
+
+// Mostly for identifying what the window/control is for debugging but could be used normally
+// These can't be used in the WindowProc user callbacks though
+_Success_(return != nullptr)
```

Comment:
I should have added it can be used with `FindWindow `to avoid globals.
Was thinking of making one to get the `Window Class` name in the same fashion but most of the dialogs don't have one